From 2270749e0849ebb65ab2c94a5ee0d2a7b2f0b2a2 Mon Sep 17 00:00:00 2001
From: Anders Darander <anders@chargestorm.se>
Date: Tue, 5 Apr 2011 20:24:54 +0200
Subject: [PATCH] Add NL_CFLAGS and _GNU_SOURCE

The NL_CFLAGS and -D_GNU_SOURCE is added to quite a few Makefiles...
This makes lowpan-tools compile using the static libnl1.

* some parts do not find the header files of libnl, thus add NL_CFLAGS
* 'struct ucred' in glibc header files is hidden under _GNU_SOURCE

Signed-off-by: Anders Darander <anders@chargestorm.se>
---
 iplink/Makefile.am |    1 +
 lib/Makefile.am    |    1 +
 src/Makefile.am    |    4 ++--
 tests/Makefile.am  |    2 +-
 4 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/iplink/Makefile.am b/iplink/Makefile.am
index bbed7ba..d563b37 100644
--- a/iplink/Makefile.am
+++ b/iplink/Makefile.am
@@ -4,3 +4,4 @@ noinst_HEADERS = iplink.h
 iproute_LTLIBRARIES = link_wpan.la
 link_wpan_la_LDFLAGS = -module -avoid-version -shared
 
+link_wpan_la_CFLAGS = $(NL_CFLAGS) -D_GNU_SOURCE
diff --git a/lib/Makefile.am b/lib/Makefile.am
index 5e15018..6936ebb 100644
--- a/lib/Makefile.am
+++ b/lib/Makefile.am
@@ -3,3 +3,4 @@ include $(top_srcdir)/Makefile.common
 noinst_LTLIBRARIES = libcommon.la
 libcommon_la_SOURCES = printbuf.c genl.c parse.c shash.c logging.c nl_policy.c
 
+libcommon_la_CFLAGS = $(NL_CFLAGS) -D_GNU_SOURCE
diff --git a/src/Makefile.am b/src/Makefile.am
index 555e214..46c4017 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -20,11 +20,11 @@ iz_SOURCES = iz.c iz-common.c iz-mac.c iz-phy.c
 noinst_HEADERS = iz.h
 
 izcoordinator_SOURCES = coordinator.c
-izcoordinator_CFLAGS = $(AM_CFLAGS) $(NL_CFLAGS) -DLEASE_FILE=\"$(leasefile)\"
+izcoordinator_CFLAGS = $(AM_CFLAGS) $(NL_CFLAGS) -DLEASE_FILE=\"$(leasefile)\" -D_GNU_SOURCE
 izcoordinator_CFLAGS += -DPID_FILE=\"$(pidfile)\"
 izcoordinator_LDADD = ../addrdb/libaddrdb.la $(LDADD) $(NL_LIBS) $(LEXLIB)
 
-iz_CFLAGS = $(AM_CFLAGS) $(NL_CFLAGS)
+iz_CFLAGS = $(AM_CFLAGS) $(NL_CFLAGS) -D_GNU_SOURCE
 iz_LDADD = $(LDADD) $(NL_LIBS)
 
 izattach.8: $(izattach_SOURCES) $(top_srcdir)/configure.ac
diff --git a/tests/Makefile.am b/tests/Makefile.am
index d491be5..00621ca 100644
--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -4,6 +4,6 @@ zbtest_PROGRAMS = listen-packet test test2 test3 test4 test5 test6 test7 gnl izl
 
 izlisten_SOURCES = listen.c
 
-gnl_CFLAGS = $(AM_CFLAGS) $(NL_CFLAGS)
+gnl_CFLAGS = $(AM_CFLAGS) $(NL_CFLAGS) -D_GNU_SOURCE
 gnl_LDADD = $(LDADD) $(NL_LIBS)
 
-- 
1.7.4.1

